pyproject.toml

```diff
@@ -525,9 +536,12 @@ ignore = [
     "SIM201",  # Use `backup_path.suffix != ".backup"` instead of `not backup_path.suffix == ".backup"`
     "SIM401",  # Use `property["items"].get("format", None)` instead of an `if` block
     "SIM910",  #  Use `data.get("identifier")` instead of `data.get("identifier", None)`
+    "TC006",   # [*] Add quotes to type expression in `typing.cast()`
+    "UP006",   # Use `dict` instead of `Dict` for type annotation
     "UP012",   # Unnecessary call to encode as UTF-8
     "UP018",   # Unnecessary {literal_type} call (rewrite as a literal)
     "UP031",   # Use format specifiers instead of percent format
+    "UP035",   # `typing.List` is deprecated, use `list` instead
```

Comment:
Would be good to fix `UP035`  and `UP006` right after this PR, I hope ruff is able to automatically convert everything for us
Ideally we should also convert `Optional[XX]` into `XX | None`, this will create a lot of changes but since there aren't that many changes between `stable` and `develop` right now this is our opportunity to do it

Reply:
There used to be some problems with Pydantic and using `XX | None` instead of `Optional` but I think that might have been addressed now.

Reply:
It worked in the SDK and I think the main issues where related to Python 3.9 so we should be good